cpp/include/cudf/aggregation.hpp

```diff
@@ -308,27 +343,44 @@ template <typename Base = aggregation>
 std::unique_ptr<Base> make_merge_sets_aggregation(null_equality nulls_equal = null_equality::EQUAL,
                                                   nan_equality nans_equal = nan_equality::UNEQUAL);
 
-/// Factory to create a LAG aggregation
-template <typename Base = aggregation>
-std::unique_ptr<Base> make_lag_aggregation(size_type offset);
-
-/// Factory to create a LEAD aggregation
+/**
+ * @brief Factory to create a MERGE_VARIANCES aggregation
+ *
+ * This aggregation is designed specificly to perform distributed computing of `VARIANCE`
+ * aggregation. The partial results input to this aggregation is generated by two groupby
+ * aggregations: `VARIANCE` and `COUNT_VALID`.
+ *
+ * In order to use this aggregation, the `aggregation_request` array input to `groupby::aggregate`
+ * must contain at least two requests:
+ *  - A request for `COLLECT_LIST` aggregation to collect the partial results of `COUNT_VALID`
+ *  - This `MERGE_VARIANCES` request, which must be given AFTER the request above so that it can
+ *    access the cached results generated by that request
+ *
+ * For a merging operation that is not a final merge (i.e., its outputs will be used as input to
+ * perform another `MERGE_VARIANCES` aggregation), a `SUM` aggregation must also be added to the
+ * same request for `COLLECT_LIST` above to produce the merged values for `COUNT_VALID`.
```

Comment:
This doesn't sound "easy to use and hard to misuse".

Reply:
Sorry, it's outdated usage doxygen. The latest usage doc is in the description of this PR.